### PR TITLE
fix(Core): disable StatusIcon ColorOverlay if no color is supplied

### PR DESCRIPTION
**Pull request by PascalPrecht**
This is needed to have the underlying SVG's color bleed through and
make use of the image's opacity at the same time.

Using `ColorOverlay` together with (half) transparent colors doesn't
work very well as it makes the underlying SVG color come through,
result in wrong colors.

Applying `opacity` on the image itself cause the `ColorOverlay` to
reduce in opacity as well. So in order to work with transparent
colors, we need a way to turn of the ColorOverlay and work with the
Image's opacity, which is what this change enables.

### CODE REVIEW

File: src/StatusQ/Core/StatusIcon.qml [hunk 1]
```diff
@@ -3,7 +3,7 @@ import QtGraphicalEffects 1.13
 
 Image {
     property string icon: ""
-    property color color
+    property string color: ""
```
Comment (PascalPrecht) [inline src/StatusQ/Core/StatusIcon.qml:6]: It seems `property color color` does get a default value that is not `undefined` or `""` even when turned into a string using `toString()`

We need an uninitialized color so changing this to `string` helps.

@borismelnik lemme know if there's a better way to check for `!!color` if `color` is of type `color`

File: src/StatusQ/Core/StatusIcon.qml [hunk 2]
```diff
@@ -19,7 +19,7 @@ Image {
     }
 
     ColorOverlay {
-        visible: statusIcon.color !== undefined
+        visible: !!statusIcon.color
```
Comment (borismelnik) [inline src/StatusQ/Core/StatusIcon.qml:22]: i think this maybe more pretty:
`layer.enabled: !!statusIcon.color
  layer.effect: ColorOverlay {
      color: statusIcon.color
  }`

File: src/StatusQ/Core/StatusIcon.qml (hunk 1 above)
Comment (borismelnik) [reply src/StatusQ/Core/StatusIcon.qml:6]: i think we can use "transparent" color by default and check it with `Qt.colorEqual(color, "transaprent")` bc image cant e invisible)))no ways in docs about it :(

Comment (PascalPrecht) [reply src/StatusQ/Core/StatusIcon.qml:6]: That worked perfectly - thanks!!

File: src/StatusQ/Core/StatusIcon.qml (hunk 2 above)
Comment (PascalPrecht) [reply src/StatusQ/Core/StatusIcon.qml:22]: Oh nice, yes, will do that

**Comment by PascalPrecht** [comment]
This has landed as https://github.com/status-im/StatusQ/commit/f1e34e39b9f6a6e2987367b1c4c02a3ae458bdb7

**Pull request closed without merging**